### ISSUE TITLE
Compilation error on GCC 11.1.0

### ISSUE THREAD

**Issue by Erihel** [colobot/colobot#1450]
Can't compile master branch or version 0.1.12 with GCC 11.1.0 (Arch Linux)

> /home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp: In static member function ‘static CBot::CBotInstr* CBot::CBotExprLitString::Compile(CBot::CBotToken*&, CBot::CBotCStack*)’:
/home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp:118:59: error: expected unqualified-id before ‘&’ token
  118 |                             catch (const std::out_of_range& e)
      |                                                           ^
/home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp:118:59: error: expected ‘)’ before ‘&’ token
  118 |                             catch (const std::out_of_range& e)
      |                                   ~                       ^
      |                                                           )
/home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp:118:59: error: expected ‘{’ before ‘&’ token
/home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp:118:61: error: ‘e’ was not declared in this scope
  118 |                             catch (const std::out_of_range& e)

This one looks like the one mentioned in #1437

>In file included from /home/erihel/Projects/colobot/src/ui/controls/edit.cpp:38:
/home/erihel/Projects/colobot/src/script/script.h:81:89: error: ‘numeric_limits’ is not a member of ‘std’
   81 |     static void ColorizeScript(Ui::CEdit* edit, int rangeStart = 0, int rangeEnd = std::numeric_limits<int>::max());
      |                                                                                         ^~~~~~~~~~~~~~
/home/erihel/Projects/colobot/src/script/script.h:81:104: error: expected primary-expression before ‘int’
   81 |     static void ColorizeScript(Ui::CEdit* edit, int rangeStart = 0, int rangeEnd = std::numeric_limits<int>::max());
      |                                                                                                        ^~~
make[2]: *** [src/CMakeFiles/colobotbase.dir/build.make:2246: src/CMakeFiles/colobotbase.dir/ui/controls/edit.cpp.o] Błąd 1

**Comment by Erihel** [comment]
clang 12.0.1 has the same errors

> /home/erihel/Projects/colobot/src/CBot/CBotInstr/CBotExprLitString.cpp:118:47: error: no type named 'out_of_range' in namespace 'std'
                            catch (const std::out_of_range& e)
                                         ~~~~~^

> /home/erihel/Projects/colobot/src/script/script.h:81:89: error: no member named 'numeric_limits' in namespace 'std'
    static void ColorizeScript(Ui::CEdit* edit, int rangeStart = 0, int rangeEnd = std::numeric_limits<int>::max());
                                                                                   ~~~~~^
/home/erihel/Projects/colobot/src/script/script.h:81:107: error: expected '(' for function-style cast or type construction
    static void ColorizeScript(Ui::CEdit* edit, int rangeStart = 0, int rangeEnd = std::numeric_limits<int>::max());
                                                                                                       ~~~^
/home/erihel/Projects/colobot/src/script/script.h:81:110: error: no member named 'max' in the global namespace
    static void ColorizeScript(Ui::CEdit* edit, int rangeStart = 0, int rangeEnd = std::numeric_limits<int>::max());

**Comment by Erihel** [comment]
This was fixed in f3ae4b420 and fc2bd6887. Please make a release or backport those two commits to master and newest version branch. Currently only dev branch compiles.

**Comment by Erihel** [comment]
New release fixed this issue, thank you.
